File: application/src/main/java/whenyourcar/application/facade/car/CarCommonFacade.java
package whenyourcar.application.facade.car;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import whenyourcar.application.dto.car.search.CarCommonResponse;
import whenyourcar.application.service.car.CarCommonService;

import java.util.Date;

@Service
@RequiredArgsConstructor
public class CarCommonFacade {
    private final CarCommonService carCommonService;

    public Page<CarCommonResponse.SearchResponseDto> searchCars(Pageable pageable, Date minAge, Date maxAge, Integer minMileage, Integer maxMileage, Integer minPrice, Integer maxPrice, String color) {
        return carCommonService.searchCarsService(pageable, minAge, maxAge, minMileage, maxMileage, minPrice, maxPrice,color);
    }

    public CarCommonResponse.SearchDescriptionResponseDto searchDescription(Long carId) {
        return carCommonService.searchDescriptionService(carId);
    }

    public Page<CarCommonResponse.SearchDetailResponseDto> searchDetailCars(Pageable pageable, String manu, String model, String submodel, String grade ) {
        return carCommonService.searchDetailCarsService(pageable, manu, model, submodel, grade);
    }

}

File: application/src/main/kotlin/howmuchcar/application/facade/car/CarCommonFacade.kt
package howmuchcar.application.facade.car

import howmuchcar.application.converter.car.CarSearchDescResponse
import howmuchcar.application.converter.car.CarSearchDetailResponse
import howmuchcar.application.converter.car.CarSearchResponse
import howmuchcar.application.port.`in`.car.CarSearchUseCase
import lombok.RequiredArgsConstructor
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service
import java.time.LocalDate

@Service
@RequiredArgsConstructor
class CarCommonFacade (
    private val carSearchUseCase: CarSearchUseCase
){
    fun searchCars(
        pageable: Pageable,
        minAge: LocalDate?,
        maxAge: LocalDate?,
        minMileage: Int?,
        maxMileage: Int?,
        minPrice: Int?,
        maxPrice: Int?,
        color: String?
    ): Page<CarSearchResponse> {
        return carSearchUseCase.searchCars(
            pageable,
            minAge,
            maxAge,
            minMileage,
            maxMileage,
            minPrice,
            maxPrice,
            color
        )
    }

    fun searchDescription(carId: Long): CarSearchDescResponse {
        return carSearchUseCase.searchCarDescription(carId)
    }

    fun searchDetailCars(
        pageable: Pageable,
        manu: String?,
        model: String?,
        submodel: String?,
        grade: String?
    ): Page<CarSearchDetailResponse> {
        return carSearchUseCase.searchCarDetail(pageable, manu, model, submodel, grade)
    }
}